File: services/typescript/fastify/src/routes/handlers/invalid_method.ts
import type { RouteHandler } from "fastify";
import { AppErrors, sendError } from "../errors.js";
import { getLogger } from "../../logger.js";

export const invalidMethodHandler: RouteHandler = (request, reply) => {
  const logger = getLogger();
  logger.warn({ method: request.method, path: request.url }, "Invalid method");
  sendError(reply, AppErrors.InvalidMethod);
};

File: services/python/starlette/src/routes/handlers/invalid_method.py
import structlog
from starlette.requests import Request
from starlette.responses import Response

from routes.errors import AppErrors, send_error


async def invalid_method_handler(request: Request, _exception: Exception) -> Response:
    structlog.get_logger().warn(
        "Invalid method", method=request.method, path=request.url.path
    )

    return send_error(AppErrors.InvalidMethod)
